vms/evm/database/wrapped_database.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"errors"
+
+	"github.com/ava-labs/libevm/ethdb"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+var (
+	_ ethdb.Batch         = (*ethBatchWrapper)(nil)
+	_ ethdb.KeyValueStore = (*ethDBWrapper)(nil)
+
+	ErrSnapshotNotSupported = errors.New("snapshot is not supported")
+	ErrStatNotSupported     = errors.New("stat is not supported")
+)
+
+func WrapDatabase(db database.Database) ethdb.KeyValueStore { return ethDBWrapper{db} }
```

Comment:
This is an adapter/wrapper pattern - I would just name this `New`. You can come up with more creative names like `FromDatabase` or `WithDatabase` to make it clear you're adapting something or deriving a type from an existing one, but since there's no other constructor defined in this package I would just go super simple and boring and go for `New` because there's nothing to disambiguate against.